llpc/util/llpcElfWriter.cpp

```diff
@@ -735,6 +737,285 @@ template <class Elf> void ElfWriter<Elf>::updateMetaNote(Context *pContext, cons
   pNewNote->data = data;
 }
 
+// =====================================================================================================================
+// Get the count of relocs in the reloc section
+//
+// @param pRelocSection : Reloc section
+template <class Elf> size_t ElfWriter<Elf>::getRelocsCount(const SectionBuffer *pRelocSection) {
+  size_t relocCount = 0;
+  if (pRelocSection->secHead.sh_size > 0) {
+    relocCount = static_cast<unsigned>(pRelocSection->secHead.sh_size / pRelocSection->secHead.sh_entsize);
+  }
+
+  return relocCount;
+}
+
+// =====================================================================================================================
+// Get the start pos of relocs used by PS in the reloc section
+//
+// @param pRelocSection : Reloc section
+// @param psIsaOffset : Ps ISA offset in the elf bin
+template <class Elf> size_t ElfWriter<Elf>::getRelocPsStartPos(const SectionBuffer *pRelocSection, size_t psIsaOffset) {
+  size_t psStartPos = 0;
+
+  size_t relocCount = getRelocsCount(pRelocSection);
+  if (relocCount > 0) {
+    auto relocs = reinterpret_cast<typename Elf::Reloc *>(const_cast<uint8_t *>(pRelocSection->data));
+    while (psStartPos < relocCount) {
+      if (relocs[psStartPos].r_offset >= psIsaOffset)
+        break;
+      psStartPos++;
+    }
+  }
+
+  return psStartPos;
+}
+
+// =====================================================================================================================
+// Create a new empty section based on the input section
+//
+// @param [in] pName : Name of the new section
+// @param [in] pInputSection : the input section
+// @param [out] pNewSection : new section returned
+template <class Elf>
+void ElfWriter<Elf>::createNewSection(const char *pName, SectionBuffer *pInputSection, SectionBuffer *pNewSection) {
+
+  // Clear the new section
+  memset(pNewSection, 0, sizeof(SectionBuffer));
```

Comment:
Does this create a new section or initialize a new section? I'm just thinking which name would be more accurate.

Reply:
When this function is called, it means there is no such section, so I think create might be more accurate.